server/routes/newspaperArchive.ts

```diff
@@ -55,7 +55,16 @@ router.get('/auth', async (_req: Request, res: Response) => {
 	);
 
 	const responseJson = (await response.json()) as NewspapersResponseBody;
-	res.redirect(responseJson.url);
+	const tpaToken = new URL(responseJson.url).searchParams.get('tpa');
```

Comment:
Should this be moved inside the if block below? It looks like we only use `tpaToken` if the `ncom-return-url` query string arg is specified.

Reply:
Good shout